File: school-registry/backend/src/main/java/com/school/registry/model/School.java
package com.school.registry.model;

import jakarta.persistence.*;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.Pattern;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.AllArgsConstructor;

@Entity
@Table(name = "schools")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class School {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @NotBlank(message = "School name is required")
    @Column(nullable = false)
    private String name;

    @NotBlank(message = "EDRPOU is required")
    @Pattern(regexp = "\\d{8}", message = "EDRPOU must be 8 digits")
    @Column(nullable = false, unique = true)
    private String edrpou;

    @NotBlank(message = "Region is required")
    @Column(nullable = false)
    private String region;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private SchoolType type;

    @Column(name = "is_active", nullable = false)
    private Boolean isActive = true;
} 

File: school-registry/frontend/src/types/School.ts
export enum SchoolType {
  GYMNASIUM = "GYMNASIUM",
  LYCEUM = "LYCEUM",
  GENERAL_SECONDARY = "GENERAL_SECONDARY",
}

export interface School {
  id: number;
  name: string;
  edrpou: string;
  region: string;
  type: SchoolType;
  isActive: boolean;
}

export interface SchoolCreateRequest {
  name: string;
  edrpou: string;
  region: string;
  type: SchoolType;
}

export const schoolTypeLabels: Record<SchoolType, string> = {
  [SchoolType.GYMNASIUM]: "Гімназія",
  [SchoolType.LYCEUM]: "Ліцей",
  [SchoolType.GENERAL_SECONDARY]: "ЗЗСО",
};
